src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -164,11 +164,18 @@ public static void AddTextDocumentServices(this IServiceCollection services)
         services.AddHandler<RazorLanguageQueryEndpoint>();
     }
 
-    public static void AddOptionsServices(this IServiceCollection services)
+    public static void AddOptionsServices(this IServiceCollection services, RazorLSPOptions currentOptions)
     {
         services.AddSingleton<IConfigurationSyncService, DefaultRazorConfigurationService>();
-        services.AddSingleton<RazorLSPOptionsMonitor>();
-        services.AddSingleton<IOptionsMonitor<RazorLSPOptions>, RazorLSPOptionsMonitor>();
+        services.AddSingleton(s =>
```

Comment:
No idea if its possible, but would it be easier to put `services.AddSingleton<RazorLSPOptions>(s => currentOptions)` at the start of this method, and would the services collection then do the right thing?

Reply:
I think that might be bad just because the options might be used by other things that meant to get the latest options instead. I could imagine doing accidental DI and it "just working" but not correctly